libsel4vm/src/arch/arm/vgic/vgic_v2.c

```diff
@@ -58,7 +58,8 @@
 #include "gicv2.h"
 #include "vdist.h"
 
-
+static uintptr_t vcpu_addr;
+static uintptr_t cpu_addr;
```

Comment:
At this point, yes, it feels like this could be reworked to avoid the globals. Since you've already done some work with that in the PR you referenced, do you want to handle that?

Reply:
I'd try to rebase my PR on this then.

Reply:
My preference would be to avoid adding the globals in this PR. The vgic_dist_device struct is intended for storing these exact mapping addresses and so they should be added there. As they aren't the same for the gic_v3, it's fine to create a different vgic_dist_device that's specialized to the gicv2 as it's private to this compilation unit.